### ISSUE TITLE
Hourly consumption , proposed solution

### ISSUE THREAD

**Issue by vingerha** [m4dm4rtig4n/enedisgateway2mqtt#95]
**Type d'installation**
Docker, HA
**Version : 0.7.7 **

I would like to propose a solution to bring hourly data to HA as well. 
Goal is to present a graph with the hourly values in HA with as main reason to show peaks during the day.
Challenge 1: the current setup does not seem to (?) extract the latest hourly values, i.e. the SQLite shows till yesterday midnight.
Challenge 2 : graph display on configurable x-axis (time)
I started with the graph setup and the only way to display custom x-axis values is with apexcharts, this can call on the attributes using "data-generator"
Based on that, I modified main.py by adding a 'hourly' sensor creator (hourly.py). The sensor stores values 5 days in the past, ... the 5 days is now hard-coded but could be made a parameter I guess.
On Challenge 1: possibly this can be improved as I am extracting data throughout the day and should be getting the very recent ones.
Aside the below files, it requires an entry on the config.yaml , section HA "hourly: true"

Please have a look below and let me know your thoughts. I will continue to work with this and possibly finetune over time.

[main.txt](https://github.com/m4dm4rtig4n/enedisgateway2mqtt/files/7828331/main.txt)
[hourly.txt](https://github.com/m4dm4rtig4n/enedisgateway2mqtt/files/7828333/hourly.txt)
[apexchart_example.txt](https://github.com/m4dm4rtig4n/enedisgateway2mqtt/files/7828506/apexchart_example.txt)

**Comment by cvalentin-dkt** [comment]
You can create Pull Request plz ?

**Comment by m4dm4rtig4n** [comment]
Add in 0.7.8

**Comment by vingerha** [comment]
I have included all of this in my pull request ... and more, a.o. it needed quite a bit of finetuning to not kick off the Enedis calls for "daily" and "detail" if data was already present in the database or of 'today'. 
At present it timely checks if the max_days_date is not reached and it also extracts the details, for that I had to adapt the "FORCE WEEK section in detail, now analysying the db for 'today 00:10:00'
Note that as long the PR remains open, it I will add/improve to that one.
